xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/flamingo.js

```diff
@@ -40,14 +40,78 @@ require(['jquery', 'bootstrap'], function($) {
         }
       }
     });
-
-    // When the drawer is close, collapse sub items
-    $(body).on('drawer.closed', function() {
-      $('.drawer-menu-sub-item').removeClass('in').attr('aria-expanded', 'false');
-    });
-
     // Activate the popover when hovering the Translate button.
     var translateButton = $('#tmTranslate [data-toggle="popover"]');
     translateButton.attr('title', translateButton.attr('data-title')).popover();
   });
 });
+
+/*
+  Handle the behavior of all drawers on the page.
+  In order to create a drawer that is compatible with this script, two elements are necessary:
+  * A drawer toggler button, that will have:
+   ** a reference to the drawer container in its attribute 'aria-control'
+   ** the class 'drawer-toggle'
+  * A drawer container, that will have:
+   ** a unique ID
+   ** the class 'drawer-nav' (for style) 
+   ** an accessible name 
+   ** The content to display inside.
+  In addition, the drawer can contain a `.drawer-close` button that will close the drawer on click.
+  For an example of drawer creation, see #tmDrawerActivator and #tmDrawer.
+ */
+require(['jquery'], function($) {
+  $('.drawer-toggle').each(function(index) {
+    // Setting up the drawer.
+    let drawerContainerToggler = $(this);
+    let drawerId = drawerContainerToggler.attr('aria-controls');
+    let drawerContainer = $(document.getElementById(drawerId));
+    
+    let openDrawer = () => drawerContainer.trigger('drawer' + index + '.opened');
+    let closeDrawer = () => drawerContainer.trigger('drawer' + index + '.closed');
+    drawerContainerToggler.on('click', openDrawer);
```

Comment:
It confuses me when reading this code that a "toggler" just opens the drawer and I looked for the place in the code where this listener is replaced to close the drawer once it has been opened. It then occurred to me that this button cannot be clicked when the drawer is open, and thus it is really just an opener. Maybe this could be made more explicit?

Reply:
It's an habit I got from working on dropdowns on menus. You're right, it doesn't make the most sense here.
Fixed in https://github.com/xwiki/xwiki-platform/pull/2282/commits/e5d890aac88a76a839ef974d01309931c972a9e5 :+1: